src/blockchain_utilities/blockchain_stats.cpp

```diff
@@ -46,6 +46,77 @@ using namespace cryptonote;
 
 static bool stop_requested = false;
 
+static bool do_inputs, do_outputs, do_ringsize, do_hours, do_emission, do_fees, do_diff;
+
+static struct tm prevtm, currtm;
+static uint64_t prevsz, currsz;
+static uint64_t prevtxs, currtxs;
+static uint64_t currblks;
+static uint64_t h;
+static uint64_t totins, totouts, totrings;
+static boost::multiprecision::uint128_t prevemission, prevfees;
+static boost::multiprecision::uint128_t emission, fees;
+static boost::multiprecision::uint128_t totdiff, mindiff, maxdiff;
+
+#define MAX_INOUT	0xffffffff
+#define MAX_RINGS	0xffffffff
+
+static uint32_t minins = MAX_INOUT, maxins;
+static uint32_t minouts = MAX_INOUT, maxouts;
+static uint32_t minrings = MAX_RINGS, maxrings;
+static uint32_t io, tottxs;
+static uint32_t txhr[24];
+
+static void doprint()
```

Comment:
I don't like how `doprint()` is now a singleton, but the logic is correct. I verified the code body matches with `diff -b`. Zero-initialization is not needed for `static` data, so that is correct AFAIK. Good patch!

Reply:
Yeah, unlike what OOP proponents would have you believe, not everything needs to be a class. Abstraction for simplifying complex systems makes sense. Abstracting something that's already braindead simple is just a pessimization.